plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -188,11 +207,17 @@ func (h *HTTPListenerV2) createHTTPServer() *http.Server {
 
 // Stop cleans up all resources
 func (h *HTTPListenerV2) Stop() {
+	if h.server != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), serverShutdownTimeout)
+		defer cancel()
+		h.server.Shutdown(ctx)
+	}
```

Comment:
I believe this should be closed after the listener. Not a huge deal but I believe we should first close the listener to prevent incoming connections and then gracefully shutdown the server.

Reply:
`Shutdown` closes the listener on its own, so this is unnecessary to begin with. I removed it.